### ISSUE TITLE
Intensities in are normalized by values in first frame

### ISSUE THREAD

**Issue by batson** [napari/napari#69]
## 🐛 Bug

If the first slice of a z-stack is all zeroes, then subsequent slices appear all white. I believe that intensities are being normalized to the range of the first slice.

## To Reproduce

```
from napari_gui.elements import Window

img = np.random.randn(100, 100, 2)
img[:,:,0] = 0
# img[:,:,0] = img[:,:,0]*0.1
# img[:,:,1] = 0

win = Window()
viewer = win.add_viewer()
layer = viewer.add_image(img, dict(itype='mono'))
layer.cmap = 'grays'
win.show()
```

## Expected behavior

Intensity scaling should be done for the whole stack at once (or perhaps as a per-slice option), not based on the first slice.

## Environment

 - napari-gui Version (e.g., 1.0): 0.0.4+14.g9a08159
 - OS X
 - Python version: 3.6.4

**Comment by kne42** [comment]
Thanks for reporting this bug @batson! Reproduced on [`9a08159`](https://github.com/Napari/napari-gui/tree/9a081595b3d51ed2e85e5119b58e1b9137b6bb77).

Summary of bug:
- `clim` is defaulted to `'auto'`
- first slice is prepared for texture upload
- `clim` is resolved from `'auto'` to the `min, max` of the first slice
- subsequent slices have this `clim` carried over (in this case `0.0, 0.0`)

We can fix this by intercepting `clim` before `vispy` gets to it and checking if `'auto'` is an option, then setting it to `np.min(image), np.max(image)`. This should additionally be reset upon `image` change.

@AhmetCanSolak is working on a PR to integrate `clim` and `gamma` into the sliders. If it's close enough to being done, we can shove this onto it, otherwise this _should_ be a simple enough PR to get done in the next few days.

**Comment by sofroniewn** [comment]
This bug has now been fixed with #85. We now calculate and set clims using the max and min of the whole image volume on loading.
